src/new/wallet.js

```diff
@@ -0,0 +1,441 @@
+      updatedBalanceMap[tokenUid] = totalBalance;
+    }
+    const newTokensHistory = Object.assign({}, state.tokensHistory, updatedHistoryMap);
+    const newTokensBalance = Object.assign({}, state.tokensBalance, updatedBalanceMap);
+
+    storage.setItem('local:tokens:history', newTokensHistory);
+    storage.setItem('local:tokens:balance', newTokensBalance);
+    */
+  };
+
+  /**
+   * Send a transaction with multi tokens
+   * @param {Object} data Array of {'address', 'value', 'token'}
+   */
+  sendMultiTokenTransaction(data) {
+    const txData = {
+      inputs: [],
+      outputs: []
+    }
+
+    const walletData = wallet.getWalletData();
+    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+
+    // First I need an array with all tokens
+    const allTokens = [];
+    for (const d of data) {
+      if (allTokens.indexOf(d.token) === -1) {
+        allTokens.push(d.token);
+      }
+    }
+
+    for (const d of data) {
+      const dataOutput = {'address': d.address, 'value': d.value, 'tokenData': tokens.getTokenIndex(allTokens, d.token.uid)};
+      const partialData = {'inputs': [], 'outputs': [dataOutput]}
+      const result = wallet.prepareSendTokensData(partialData, d.token, true, historyTxs, allTokens);
+
+      if (!result.success) {
+        console.log('Error sending tx:', result.message);
+        return;
+      }
+
+      const dataToken = result.data;
+
+      txData['inputs'] = [...txData['inputs'], ...dataToken['inputs']];
+      txData['outputs'] = [...txData['outputs'], ...dataToken['outputs']];
+    }
+
+    txData.tokens = tokens.filterTokens(allTokens, HATHOR_TOKEN_CONFIG).map((token) => token.uid);
+
+    const updateOutputs = (val) => {
+      const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+      for (const input of txData.inputs) {
+        historyTxs[input.tx_id]['outputs'][input.index]['spent_by'] = val;
+      }
+    }
+
+    // TODO XXX Setting input as spent by, so it won't be chosen again
+    updateOutputs('2');
+
+    const promise = new Promise((resolve, reject) => {
+      transaction.sendTransaction(txData, this.pinCode).then((result) => {
+        if (result.success === false) {
+          // XXX Setting outputs as spent_by null again
+          updateOutputs(null);
+        }
+        resolve(result);
+      }, (error) => {
+        // XXX Setting outputs as spent_by null again
+        updateOutputs(null);
+        reject(error);
+      });
+    })
+
+    return promise;
+  }
+
+  /**
+   * Send tokens to only one address.
+   *
+   * @param {String} address Address to send the tokens
+   * @param {number} value Amount of tokens to be sent
+   * @param {Object} token Token object {'uid', 'name', 'symbol'}. Optional parameter if user already set on the class
+   *
+   * @return {Promise} Promise that resolves when transaction is sent
+   **/
+  sendTransaction(address, value, token) {
```

Comment:
We can rename it to `sendSimpleTransaction`. But  I think the `sendMultiTokenTransaction` (renamed to `sendTransaction`) is enough and it may be removed.

Reply:
I will keep it this way until we refactor `sendMultiTokenTransaction` to a usable method.